app/uk/gov/hmrc/ndrrprotofrontend/controllers/Common.scala

```diff
@@ -0,0 +1,84 @@
+      county = Some("West Sussex"),
+      postcode = Postcode("BN110AA"))
+
+    val reference0 = Reference("987765JK99")
+
+    val address1 = VoaAddress(
+      line1 = "87a",
+      line2 = Some("High St"),
+      town = "Hythe",
+      county = Some("Kent"),
+      postcode = Postcode("HY270AA")
+    )
+
+    val reference1 = Reference("WillIAm")
+
+    val row0 = VoaRow(Seq(address0, reference0, TrnNumber(75)))
+    val row1 = VoaRow(Seq(address1, reference1, TrnNumber(75)))
+
+    VoaTable(headers, Seq(row0, row1))
+
+  }
+
+  def makeSummaryList(row: Seq[VoaSummaryListRow], card: Option[VoaCard] = None)(implicit messages: Messages): Aliases.SummaryList = buildSummaryList(row,card)
+
+  val address1 = VoaAddress(
+    line1 = "87a",
+    line2 = Some("High St"),
+    town = "Hythe",
+    county = Some("Kent"),
+    postcode = Postcode("HY270AA")
+  )
+  val fullName: FullName = FullName("Sarah",None,"Philips")
+  val email:String = "olivia@thecornershop.co.uk."
+  val dateOfBirth: DateOfBirth = DateOfBirth(5,"January", 1978)
+  val row1: VoaSummaryListRow = VoaSummaryListRow("voa.name.title",Seq(fullName.toString),Some(Link(url = "some-href", linkId =  "linkId", messageKey ="voa.change.link")))
+  val row2 = VoaSummaryListRow("voa.dob.title",Seq(dateOfBirth.toString),Some(Link(url = "some-href", linkId ="linkId", messageKey ="voa.change.link")))
+  val row3 = VoaSummaryListRow("voa.address.title",Seq(address1.line1,address1.town,address1.postcode.toString),Some(Link(url ="some-href", linkId ="linkId", messageKey ="voa.change.link")))
+  val row4 = VoaSummaryListRow("voa.contactInformation.title",Seq.empty,Some(Link(url = "some-href", linkId = "linkId", messageKey ="voa.contactInformation.enterContact")))
+  val card = VoaCard("Contact Info",Some(Seq(link1,link2)))
+  private lazy val link1 = Link(url = "some-href", linkId =  "linkId", messageKey ="voa.contactInformation.enterContact")
+  private lazy val link2: Link = Link(url = "some-href", linkId =  "linkId", messageKey ="voa.contactInformation.enterContact")
+
+  def makeAVisibleTable()(implicit messages: Messages): Table = createVoaTable().buildTable()
+
+  def makeSummaryListRow(row: Seq[VoaSummaryListRow])(implicit messages: Messages): Aliases.SummaryList = buildSummaryList(row,None)
+  def contactName(name: String) =   VoaSummaryListRow("registration.checkAnswers.contactDetails.table.name",Seq(name),Some(Link(url ="some-href",messageKey = "voa.change.link", linkId = "name-Id")))
+  def emailAddress(email:String) =  VoaSummaryListRow("registration.checkAnswers.contactDetails.table.emailAddress",Seq(email),Some(Link(url ="/ndrr-proto-frontend/registration/email",messageKey = "voa.change.link", linkId = "email-Id")))
+  def phoneNumber(number: String) =   VoaSummaryListRow("registration.checkAnswers.contactDetails.table.phoneNumber",Seq(number),Some(Link(url ="/ndrr-proto-frontend/registration/phone-number",messageKey = "voa.change.link", linkId = "phone-number-Id")))
```

Comment:
With each of these links, instead of passing in the link as a string, I would change the type to Call, then it would be controllers.routes.registration.email.url etc. I think that's a cleaner way of doing it rather than hardcoding in the url.

Reply:
If you want to pass say "somehref" then all you'd have to do is Call("GET","some-href")